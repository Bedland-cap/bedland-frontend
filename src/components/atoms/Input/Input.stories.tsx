import type { Meta, StoryObj } from '@storybook/react';
import { FieldValues, UseFormRegister } from 'react-hook-form';
import GlobalStyle from 'theme/globalStyles';
import ThemeProvider from 'theme/ThemeContext';
import Input from './Input';
import { IInput } from './Input.types';

const meta = {
  title: 'atoms/Input',
  component: Input,
  tags: ['autodocs'],
  decorators: [
    (Story) => (
      <ThemeProvider>
        <GlobalStyle />
        {Story()}
      </ThemeProvider>
    ),
  ],
} satisfies Meta<typeof Input>;

export default meta;
type Story = StoryObj<typeof meta | IInput>;

export const Default: Story = {
  args: {
    input: 'Name',
    type: 'Name',
    placeholder: 'Name',
    register: {} as UseFormRegister<FieldValues>,
  },
};
