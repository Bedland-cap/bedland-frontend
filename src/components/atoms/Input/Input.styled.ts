import styled from 'styled-components';

const LoginInput = styled.input<{ borderColor: string }>`
  border: 1px solid ${(props) => props.borderColor};
  border-radius: 0.5rem;
  display: flex;
  height: 2.875rem;
  padding: 0.625rem 1rem 0.625rem 1rem;
  width: 19.875rem;
  &:disabled,
  &:hover,
  &:active,
  &:focus {
    outline: none;
  }
`;

export default LoginInput;
