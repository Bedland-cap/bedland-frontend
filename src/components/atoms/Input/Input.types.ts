import { FieldValues, UseFormRegister } from 'react-hook-form';
import { HTMLInputTypeAttribute } from 'react';

type Input = 'login' | 'password' | 'confirmPassword' | 'Name';

export type RegisterOptions = { validate: (value: string) => boolean } | null;

export interface IInput {
  input: Input;
  type: HTMLInputTypeAttribute;
  placeholder: string;
  register: UseFormRegister<FieldValues>;
  registerOptions?: RegisterOptions;
}
