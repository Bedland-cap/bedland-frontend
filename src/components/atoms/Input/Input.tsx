/* eslint-disable react/jsx-props-no-spreading */
// without changing lint rules this component will block many tests,
// and block the build itself.
import { useContext } from 'react';
import loginRegexPattern from 'utils/constants/loginRegexPatterns';
import { ThemeContext } from 'theme/ThemeContext';
import LoginInput from './Input.styled';
import { IInput } from './Input.types';

const Input = ({
  input,
  type = 'text',
  placeholder,
  register,
  registerOptions = null,
}: IInput) => {
  const { palette } = useContext(ThemeContext);

  return (
    <LoginInput
      id={input}
      type={type}
      placeholder={placeholder}
      borderColor={palette.primaryLight}
      {...register(input, { ...loginRegexPattern[input], ...registerOptions })}
    />
  );
};

export default Input;
