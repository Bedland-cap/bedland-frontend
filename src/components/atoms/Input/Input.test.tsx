import renderer from 'react-test-renderer';
import { useForm } from 'react-hook-form';
import ThemeProvider from 'theme/ThemeContext';
import Input from './Input';

const SnapshotInput = () => {
  const { register } = useForm();
  return (
    <ThemeProvider>
      <form action=''>
        <Input
          input='login'
          type='text'
          placeholder='login'
          register={register}
        />
      </form>
    </ThemeProvider>
  );
};

describe('Input', () => {
  it('renders correctly', () => {
    const snapshot = renderer.create(<SnapshotInput />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
