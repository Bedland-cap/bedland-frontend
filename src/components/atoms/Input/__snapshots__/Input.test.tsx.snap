// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
<form
  action=""
>
  <input
    className="sc-bcXHqe dMnLRc"
    id="login"
    name="login"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="login"
    type="text"
  />
</form>
`;
